### PR TITLE
[WIP] Improve targets

### PR DESCRIPTION
**Pull request by lightsprint09**
This is working in progress branch which has the following goals:

* Remove all targets
* Have one single multi platform target (read [here](http://ilya.puchka.me/xcode-cross-platform-frameworks/) and [here](http://promisekit.org/news/2016/08/Multiplatform-Single-Scheme-Xcode-Projects/))
* Remove Starscream from Source and fetch it via a dependency manager (carthage, spm, pods)

Are you interested in following the this idea?

### CODE REVIEW

**Comment by nuclearace** [comment]
Yes, I think this would be great. Granted it doesn't break travis and is still easy to debug.

One note though, I'm using a custom Starscream repo because there were like two changes that had to be made for some ObjC compatibility. Would it be possible to keep that?

**Comment by nuclearace** [comment]
Also for Carthage frameworks. Do you have to include the Checkout? Or is the .resolved and Cartfile enough?

File: Tests/SocketObjectiveCTest.m [hunk 1]
```diff
@@ -99,9 +98,9 @@ - (void)testSocketManager {
     [manager removeSocketWithLabel:@"test"];
 }
 
-- (void)testSSLSecurity {
```
Comment (nuclearace) [inline Tests/SocketObjectiveCTest.m]: This was one of the custom changes I have to make to Starscream I believe.

**Comment by nuclearace** [comment]
Also, should be opened against `development`

**Comment by lightsprint09** [comment]
Thanks for you feedback.

Maybe I should try to break this up into two separate PRs. One for cross-platform target, the other for eventually removing Starscream code.

Can you give me more information where you did these customization in Starscream

You can set the default branch to development, so all PRs will go against development by default.

**Comment by lightsprint09** [comment]
I close this for now.

**Comment by nuclearace** [comment]
@lightsprint09 I don't actually have access to the settings on this repo, otherwise I would've changed that setting.
https://github.com/nuclearace/Starscream/tree/socket.io is where I keep the changes for this project.

**Comment by nuclearace** [comment]
@lightsprint09 https://github.com/socketio/socket.io-client-swift/issues/761 was one of the reasons I moved Starscream into its own repo. I kept breaking this everytime I synced the local copy of Starscream.

**Comment by nuclearace** [comment]
Also https://github.com/socketio/socket.io-client-swift/issues/598

**Comment by nuclearace** [comment]
I tried to get those changes put into the upstream but the response was Starscream is meant for Swift, so those changes were best left out. 🤷‍♀️

**Comment by nuclearace** [comment]
Also @lightsprint09 it might make more sense to just do it on the Swift 4.0 branch. Since Swift 4 is right around the corner.

**Comment by lightsprint09** [comment]
sure.

**Pull request closed without merging**